### PR TITLE
Optimize db request

### PR DESCRIPTION
**Pull request by xlqian**
request old trip_update in bulk

### CODE REVIEW

File: kirin/core/handler.py [hunk 1]
```diff
@@ -129,10 +129,13 @@ def handle(real_time_update, trip_updates, contributor):
     if not real_time_update:
         raise TypeError()
 
+    id_timestamp_tuples = [(tu.vj.navitia_trip_id, tu.vj.get_start_timestamp()) for tu in trip_updates]
```
Comment (azime) [inline kirin/core/handler.py]:
```python
((tu.vj.navitia_trip_id, tu.vj.get_start_timestamp()) for tu in trip_updates)
```
no?

File: kirin/core/handler.py [hunk 2]
```diff
@@ -129,10 +129,13 @@ def handle(real_time_update, trip_updates, contributor):
...
+    old_trip_updates = TripUpdate.find_by_dated_vjs(id_timestamp_tuples)
     for trip_update in trip_updates:
         # find if there is already a row in db
-        old = TripUpdate.find_by_dated_vj(trip_update.vj.navitia_trip_id, trip_update.vj.get_start_timestamp())
-        # merge the theoric, the current realtime, and the new realtime
+        old = next((tu for tu in old_trip_updates if tu.vj.navitia_trip_id == trip_update.vj.navitia_trip_id
+                    and tu.vj.get_start_timestamp() == trip_update.vj.get_start_timestamp()), None)
+        # merge the base schedule, the current realtime, and the new realtime
```
Comment (azime) [inline kirin/core/handler.py]: :+1:

File: kirin/core/handler.py (hunk 1 above)
Comment (xlqian) [reply kirin/core/handler.py]: it'll be evaluated in the function which uses it... so it won't change anything if I use generator :)

File: kirin/core/model.py [hunk 3]
```diff
@@ -229,6 +229,12 @@ def find_by_dated_vj(cls, navitia_trip_id, start_timestamp):
                                               VehicleJourney.start_timestamp == start_timestamp).first()
 
     @classmethod
+    def find_by_dated_vjs(cls, id_timestamp_tuples):
+        from sqlalchemy import tuple_
+        return cls.query.join(VehicleJourney).filter(tuple_(VehicleJourney.navitia_trip_id, VehicleJourney.start_timestamp)
```
Comment (pbougue) [inline kirin/core/model.py]: Minor: Line length

File: kirin/core/model.py [hunk 4]
```diff
@@ -255,10 +261,7 @@ def remove_by_contributors_and_period(cls, contributors, start_date=None, end_da
 
     def find_stop(self, stop_id):
         #TODO: we will need to handle vj who deserve the same stop multiple times
-        for st in self.stop_time_updates:
-            if st.stop_id == stop_id:
-                return st
-        return None
+        return next((st for st in self.stop_time_updates if st.stop_id == stop_id), None)
```
Comment (pbougue) [inline kirin/core/model.py]: Minor: Not sure we need this as we will have to be smarter and probably test more things
(you can also update comment above)

Comment (xlqian) [reply kirin/core/model.py]: it does exactly the same thing, I just made it more pythonic

**Pull request merged**